### PR TITLE
xcatprobe fixes

### PR DESCRIPTION
**Pull request by gurevichmark**
### The PR is to fix issue _#6929_

Verified on RH, SLES and Ubuntu

### CODE REVIEW

File: xCAT-probe/subcmds/xcatmn [hunk 1]
```diff
@@ -173,7 +174,8 @@ sub do_main_job {
         $rst        = check_dhcp_leases(\$checkpoint, \@error);
         print_check_result($checkpoint, "w", $rst, \@error);
         $rc |= $rst;
-    } else {
+
```
Comment (besawn) [inline xCAT-probe/subcmds/xcatmn:177]: This logic needs to be modified in order to run on both MN and SN.

File: xCAT-probe/subcmds/xcatmn [hunk 2]
```diff
@@ -1005,13 +1007,30 @@ sub check_dns_service {
                     }
                 } else {
 
-                    # if there is no sn, nslookup mnip
+                    # if there is no sn, "nslookup mnip mnip"
                     my $nslkp = `nslookup -type=A $serverip $serverip 2>&1`;
                     chomp($nslkp);
                     my $tmp = grep { $_ =~ "Server:[\t\s]*$serverip" } split(/\n/, $nslkp);
                     if (!$tmp) {
                         $rc = 1;
```
Comment (besawn) [inline xCAT-probe/subcmds/xcatmn:1015]: This section should return from here rather than setting an rc and then returning at the bottom of the function.

**Review by besawn — CHANGES_REQUESTED** [review]
The team conducted a group review of this PR and identified two changes that are needed.

**Comment by gurevichmark** [comment]
Another commit added to fix issues identified during the review and grammar and spelling fixes.

**Review by besawn — APPROVED** [review]

**Pull request merged**